drivers/platform/stm32/stm32_hal.h

```diff
@@ -0,0 +1,7 @@
+#ifndef _STM32_HAL_H
```

Comment:
What is the purpose of this file?

Reply:
- it includes an auto-generated file, main.h which in turn includes the chip specific stm32f4xx_hal.h file (dependant on the MCU family)
- it exposes the SystemClock_Config() function which is also autogenerated in a main.c file and is not otherwise exposed. This SystemClock_Config() function is very important, it initializes the core clock, prescalers, PLLs etc. according to what the user set in the .ioc file.